### ISSUE TITLE
GigaSpeech: 1% absolute WER discrepancy between training script eval and evaluation through StreamingASR

### ISSUE THREAD

**Issue by asumagic** [speechbrain/speechbrain#2753]
### Describe the bug

When evaluating the Conformer-Transducer model on GigaSpeech (related: #2752), I observed a major discrepancy (12.0% WER -> 10.9% WER) between the regular evaluation logic (inside the training script) and using [**a custom script that evaluates using `StreamingASR`**](https://gist.github.com/asumagic/66aaf43156c31866cf2ebcffda46ea01). See:

```
%WER 11.97 [ 46777 / 390744, 7308 ins, 12193 del, 27276 sub ]
%WER 10.94 [ 42749 / 390744, 7734 ins, 8980 del, 26035 sub ]
```

Notice that the `StreamingASR` path increases insertions, but greatly reduces deletion count.

For that test, I emulated full context by setting a huge chunk size.

This is a really significant difference. I wonder if it affects other datasets we are testing against with this model.

This might indicate:
- differences in padding? (streaming logic does some amount of left-padding -- would explain reduced deletions)
- a masking bug of some sort? (not sure where that would be)

The detailed WER outputs are listed below (inferencepath uses the custom script):
- [test_wer.txt](https://github.com/user-attachments/files/17679031/test_wer.txt)  
- [test-wer-inferencepath.txt](https://github.com/user-attachments/files/17679026/test-wer-inferencepath.txt)

### Expected behaviour

The discrepancy should be much smaller.

**Comment by asumagic** [comment]
Also it largely doesn't seem to care about batch size for the StreamingASR script, as a batch size of 1 yields very close results to a batch size of 32 even with the bogus `lengths` tensor. That said the test set seems sorted in size so the impact is limited.
